src/win/fs.c

```diff
@@ -1940,6 +1940,166 @@ INLINE static void fs__stat_prepare_path(WCHAR* pathw) {
   }
 }
 
+INLINE static int fs__stat_split_path(const WCHAR* filename, WCHAR** dir,
```

Comment:
Is this code basically a copy of `uv__split_path` in fs-event.c? If so, it should be factored out elsewhere, not copied here...

Reply:
Yes, it was a copy of it. I've added the definition of the original implementation to `internal.h` .  I hope I put it in the right place.